engine/src/main/java/org/hibernate/validator/metadata/BeanMetaDataClassNormalizer.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.metadata;
+
+import org.hibernate.validator.Incubating;
+
+/**
+ * Defines how the validated class is normalized before being used as the reference to get the bean metadata.
```

Comment:
I don't really understand the purpose of this contract :( Given it's public I think this should be improved to make the usage more apparent. Perhaps an example added to the JavaDoc would help? Also the method should be documented.